File: WebApp/src/main/webapp/app/service/CompanyService.js
Ext.define("TransDocs.service.CompanyService",{
    extend: 'Ext.Base',
    singleton: true,

    requires: [
        "TransDocs.model.dictionary.CompanyModel",
        "TransDocs.data.store.dictionary.CompanyStore"
    ],

    currentCompany: null,

    constructor: function () {
        Ext.data.NodeInterface.decorate("TransDocs.model.dictionary.CompanyModel");
        this.callParent(arguments);
        var companyStore = Ext.create("TransDocs.data.store.dictionary.CompanyStore");
        var me = this;
        companyStore.load({
            id: 'current',
            callback:function(records, operation, success){
                if(success) {
                    me.currentCompany = records[0];
                }else{
                    throw "current company not loaded";
                }
            }
        })
    },

    getCurrentCompany: function(){
        if(!this.currentCompany){
            throw "current company not found";
        }
        return this.currentCompany;
    },

    newCompany: function(session){
        var company = session.createRecord("TransDocs.model.dictionary.CompanyModel");
        return company;
    }
});

File: Service/src/main/java/com/td/service/crud/dictionary/company/CompanyService.java
package com.td.service.crud.dictionary.company;

import com.td.model.repository.dictionary.CompanyRepository;
import com.td.model.entity.dictionary.company.CompanyModel;
import com.td.service.crud.CRUDService;
import com.td.service.crud.dictionary.DictionaryCRUDService;

/**
 * Created by zerotul on 25.03.15.
 */
public interface CompanyService extends DictionaryCRUDService<CompanyModel>, CRUDService<CompanyModel> {

    public CompanyModel getCurrentCompany();
}
